R/get_analysis_date.R

```diff
@@ -0,0 +1,270 @@
+#' get_analysis_date(
+#'   simu_data,
+#'   target_event_overall = 150,
+#'   target_event_per_stratum = c(100, NA))
+#'
+#' # example 5: cut for analysis when there are at least 100 events
+#' # in the biomarker positive population,
+#' # and at least 200 events in the biomarker negative population, which arrives later.
+#' # But will stop at the 30-th months if events less than 100/200.
+#' get_analysis_date(
+#'   simu_data,
+#'   target_event_per_stratum = c(100, 200),
+#'   max_extension_for_target_event = 30)
+#'
+#' # example 6: cut for analysis after 12 months followup when 80%
+#' # of the patients are enrolled in the overall population.
+#' get_analysis_date(
+#'   simu_data,
+#'   enroll_rate = enroll_rate,
+#'   min_n_overall = n * 0.8,
+#'   min_followup = 12)
+#'
+#' # example 7: cut for analysis when 12 months after at least
+#' # 200/160 patients are enrolled at the
+#' # biomarker positive/negative population.
+#' get_analysis_date(
+#'   simu_data,
+#'   enroll_rate = enroll_rate,
+#'   min_n_per_stratum = c(200, 160),
+#'   min_followup = 12)
+#' get_analysis_date(
+#'   simu_data,
+#'   enroll_rate = enroll_rate,
+#'   min_n_per_stratum = c(200, NA),
+#'   min_followup = 12)
+get_analysis_date <- function(
+    simu_data,
+    # Option 1: planned calendar time for the analysis
+    planned_calendar_time = NA,
+    # Option 2: reach targeted events
+    target_event_overall = NA,
+    target_event_per_stratum = NA,
+    # Option 3: max time extension to reach targeted events
+    max_extension_for_target_event = NA,
+    # Option 4: planned minimum time after the previous analysis
+    previous_analysis_date = 0,
+    min_time_after_previous_analysis = NA,
+    # Option 5: minimal follow-up time after specified enrollment fraction
+    enroll_rate = NA,
+    min_n_overall = NA,
+    min_n_per_stratum = NA,
+    min_followup = NA
+){
+  # input checking
+  input_check_scale <- function(x = NA, label = "x"){
+    if(!is.na(x)){
+      if(is.numeric(x) & x < 0){
+        stop(paste0(label, " must be a positive number!"))
+      } else if(!is.numeric(x)){
+        stop(paste0(label, " must be a numerical value!"))
+      }
+    }
+  }
+  input_check_scale(planned_calendar_time, label = "planned_calendar_time")
+  input_check_scale(target_event_overall, label = "target_event_overall")
+  input_check_scale(max_extension_for_target_event, label = "max_extension_for_target_event")
+  input_check_scale(min_time_after_previous_analysis, label = "min_time_after_previous_analysis")
+  input_check_scale(min_n_overall, label = "min_n_overall")
+  input_check_scale(min_followup, label = "min_followup")
+
+  input_check_vector <- function(x = NA, label = "x"){
+    if (!(all(is.na(x) | (is.numeric(x) & x > 0)))) {
+      stop(paste0(label, " must be a positive number with either NA or positive numbers!"))
+    }
+  }
+  input_check_vector(target_event_per_stratum)
+  input_check_vector(min_n_per_stratum)
```

Comment:
These two are missing the argument `label`

Reply:
Nice catch! I have just added the 2 labels in the latest commit.